### PR TITLE
Allow selfloop to have waypoints. Update docstring

### LINKED ISSUE #103: `tangents` and `waypoints` support for self-loops
It seems that `tangents` does not apply to `selfedges`. What are your thoughts on applying it to these? This way, `tangents` would apply to all edge types and we wouldn't need the `selfedge_direction` attribute. The `selfedge_direction` has a major drawback: you can only specify the direction going out of the node, but not the direction coming into the node.

We could also support other attributes like `waypoints` for selfedges.

### PR DESCRIPTION
**Pull request by hdavid16**
Resolves #103 
Also fixes bug #105

### CODE REVIEW

**Review by hexaeder — COMMENTED** [review]
might be also nice to throw some quick example together for the [refests.jl](https://github.com/MakieOrg/GraphMakie.jl/blob/master/docs/examples/reftests.jl) file .. maybe just the one you posted above

File: src/recipes.jl [hunk 1]
```diff
@@ -356,61 +364,59 @@ function find_edge_paths(g, attr, pos::AbstractVector{PT}) where {PT}
...
-
-                commands = [MoveTo(p1), CurveTo(c1, c2, p2)]
-                paths[i] = BezierPath(commands)
-            else # straight line
-                paths[i] = Path(p1, p2)
+        if !isnothing(waypoints) && !isempty(waypoints)#there are waypoints
+            # the waypoints may already include the endpoints (remove them if so)
+            waypoints[begin] == p1 && popfirst!(waypoints)
+            waypoints[end] == p2 && pop!(waypoints)
+            if p1 == p2 && isempty(waypoints)
```
[the remark this answers is not in the document]
Comment (hdavid16) [reply src/recipes.jl]: fixed

File: src/recipes.jl [hunk 2]
```diff
@@ -356,61 +364,59 @@ function find_edge_paths(g, attr, pos::AbstractVector{PT}) where {PT}
...
+                paths[i] = selfedge_path(g, pos, src(e), size, direction, width)
+            elseif isempty(waypoints) || radius === nothing || radius === :spline 
+                paths[i] = Path(p1, waypoints..., p2; tangents, tfactor)
+            elseif radius isa Real
+                paths[i] = Path(radius, p1, waypoints..., p2)
+            else
+                throw(ArgumentError("Invalid radius $radius for edge $i!"))
             end
+        elseif p1 == p2 # selfedge
```
[the remark this answers is not in the document]
Comment (hdavid16) [reply src/recipes.jl]: done

**Comment by hdavid16** [comment]
> might be also nice to throw some quick example together for the [refests.jl](https://github.com/MakieOrg/GraphMakie.jl/blob/master/docs/examples/reftests.jl) file .. maybe just the one you posted above

@hexaeder I added one at the end of the reftests.jl file.

**Review by hexaeder — APPROVED** [review]

**Pull request merged**